tests/testsCore/src/androidTest/kotlin/com/wire/android/tests/core/pages/ConversationViewPage.kt

```diff
@@ -0,0 +1,345 @@
+        val currentPackage = device.currentPackageName
+        assertFalse(
+            "❌ Wire app is still in foreground: $currentPackage",
+            currentPackage.contains("APP_")
+        )
+    }
+
+    fun typeMessageInInputField(message: String): ConversationViewPage {
+        val inputField = UiWaitUtils.waitElement(messageInputField)
+        inputField.click()
+        // Use shell command to input text
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("input text ${message.replace(" ", "%s")}")
+        return this
+    }
+
+    fun clickSendButton(): ConversationViewPage {
+        UiWaitUtils.waitElement(sendButton).click()
+        return this
+    }
+
+    fun assertMessageSentIsVisible(message: String): ConversationViewPage {
+        val messageSelector = UiSelectorParams(text = message)
+        val messageElement = UiWaitUtils.waitElement(messageSelector)
+
+        Assert.assertTrue(
+            "❌ Message '$message' is not visible in the conversation",
+            !messageElement.visibleBounds.isEmpty
+        )
+        return this
+    }
+
+    fun assertMessageNotVisible(text: String) {
+        val element = findElementOrNull(UiSelectorParams(text = text))
+
+        Assert.assertTrue(
+            "❌ Message '$text' is still visible on the screen.",
+            element == null || element.visibleBounds.isEmpty
+        )
+    }
+
+    fun assertMessageReceivedIsVisible(message: String): ConversationViewPage {
+        val messageSelector = UiSelectorParams(text = message)
+        val messageElement = UiWaitUtils.waitElement(messageSelector)
+
+        Assert.assertTrue(
+            "❌ Message '$message' is not visible in the conversation",
+            !messageElement.visibleBounds.isEmpty
+        )
+        return this
+    }
+
+    fun tapBackButtonOnConversationViewPage(): ConversationViewPage {
+        UiWaitUtils.waitElement(backButton).click()
+        return this
+    }
+
+    fun tapMessageInInputField(): ConversationViewPage {
+        val inputField = UiWaitUtils.waitElement(messageInputField)
+        inputField.click()
+        return this
+    }
+
+    fun tapSelfDestructTimerButton(): ConversationViewPage {
+        val button = UiWaitUtils.waitElement(selfDestructTimerButton)
+        button.click()
+        return this
+    }
+
+    fun assertSelfDestructOptionVisible(label: String) {
+        val option = UiWaitUtils.waitElement(selfDestructOption(label))
+        Assert.assertTrue("Self-destruct option '$label' is not visible", !option.visibleBounds.isEmpty)
+    }
+
+    fun tapSelfDestructOption(label: String) {
+        val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        val element = device.findObject(
+            UiSelector().text(label).className("android.widget.TextView")
+        )
+        element.click()
+    }
+
+    fun assertSelfDeletingMessageLabelVisible() {
```

Comment:
Similar as previous comment. In case android app refers to the button as `SelfDesctruct`, then we should change the name of this function to reflect that.